spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -745,6 +807,20 @@ private void cleanUpAfterCopy(File fileToWriteTo, File resultFile, File original
 		if (this.deleteSourceFiles && originalFile != null) {
 			originalFile.delete();
 		}
+
+		setPermissions(resultFile);
+	}
+
+	/**
+	 * Set permissions on newly written files.
+	 * @param resultFile the file.
+	 * @throws IOException any exception.
+	 * @since 5.0
+	 */
+	protected void setPermissions(File resultFile) throws IOException {
+		if (this.permissions != null) {
+			Files.setPosixFilePermissions(resultFile.toPath(), this.permissions);
```

Comment:
If there's an option to check that this operation isn't supported on the target OS and `WARN` (or even `ERROR`) end-user on the matter, but not fail anyway?

Looking to the JavaDocs we really may fail and throw an exception to the caller:
```
  * @throws  UnsupportedOperationException
     *          if the associated file system does not support the {@code
     *          PosixFileAttributeView}
     * @throws  ClassCastException
     *          if the sets contains elements that are not of type {@code
     *          PosixFilePermission}
     * @throws  IOException
     *          if an I/O error occurs
     * @throws  SecurityException
     *          In the case of the default provider, and a security manager is
     *          installed, it denies {@link RuntimePermission}<tt>("accessUserInformation")</tt>
     *          or its {@link SecurityManager#checkWrite(String) checkWrite}
     *          method denies write access to the file.
     */
    public static Path setPosixFilePermissions(Path path,
                                               Set<PosixFilePermission> perms)
```